lib/faktory/processor.rb

```diff
@@ -133,6 +133,7 @@ def dispatch(payload)
             klass  = constantize(payload['jobtype'.freeze])
             jobinst = klass.new
             jobinst.jid = payload['jid'.freeze]
+            jobinst.bid = payload['custom'.freeze].to_h['bid'.freeze]
```

Comment:
@mperham i don't really understand the usage of `.freeze` here, but i just kept it the same as the line above

Reply:
This is already done by faktory/middleware/batch, which is required by faktory -> faktory/batch.

Reply:
Hmm ok. `bid` is 100% not set when accessing from within a job.

Reply:
This isn't a good sign:

```
2020-01-20T23:18:05.165Z 65683 TID-owmy4alks DEBUG: Client Middleware: Faktory::Middleware::Batch::Client
2020-01-20T23:18:05.165Z 65683 TID-owmy4alks DEBUG: Worker Middleware: 
```

Reply:
```
Faktory.configure_worker do |config|
  config.client_middleware do |chain|
    chain.add Faktory::Middleware::Batch::Client
  end
  config.worker_middleware do |chain|
    chain.add Faktory::Middleware::Batch::Worker
  end
end
```

Adding some logging in here, and it aint getting there.

Reply:
@mperham worth noting, the middlewhere chain code is working, as i built a Honeybadger plugin and it's fine.

```
2020-01-20T23:25:02.610Z 43527 TID-oxvve19e8 DEBUG: Worker Middleware: Honeybadger::Plugins::Faktory::Middleware
```

```
::Faktory.configure_worker do |faktory|
  faktory.worker_middleware do |chain|
    chain.prepend Middleware
  end

  faktory.error_handlers << lambda do |ex, params|
    opts = {parameters: params}
    Honeybadger.notify(ex, opts)
  end
end
```